packages/core/contracts/zodiac/OptimisticGovernor.sol

```diff
@@ -0,0 +1,321 @@
+            int256(1e18)
+        );
+
+        emit TransactionsProposed(id, proposer, time, proposal);
+    }
+
+    function executeProposal(
+        uint256 _proposalId,
+        Transaction[] memory _transactions,
+        bytes memory _explanation,
+        uint32 _originalTime,
+        SkinnyOptimisticOracleInterface.Request memory _request
+    ) public payable nonReentrant {
+        // Recreate the proposal hash from the inputs and check that it matches the stored proposal hash.
+        uint256 id = _proposalId;
+
+        // Construct the ancillary data.
+        bytes memory ancillaryData = _explanation;
+        AncillaryData.appendKeyValueUint(ancillaryData, "id", id);
+
+        // This will reject the transaction if the proposal hash generated from the inputs does not match the stored proposal hash.
+        // require(proposalHashes[id] == keccak256(abi.encodePacked(proposalData)), "proposal hash does not match");
+        require(proposalHashes[id] == keccak256(abi.encode(_transactions)), "proposal hash does not match");
+
+        // Remove proposal hash so transactions can not be executed again.
+        proposalHashes[id] = 0;
+
+        // This will revert if the price has not settled.
+        (, int256 price) =
+            skinnyOptimisticOracle.settle(address(this), identifier, _originalTime, ancillaryData, _request);
+        require(price == 1e18, "Proposal was rejected");
+
+        for (uint256 i = 0; i < _transactions.length; i++) {
+            require(i == 0 || _transactions[i - 1].to == address(0), "Previous tx not yet executed");
+            Transaction memory transaction = _transactions[i];
+            require(transaction.to != address(0), "Tx already executed");
+            require(msg.value == transaction.value, "Must send exact amount of ETH");
+
+            // Delete the transaction before execution to avoid any potential re-entrancy issues.
+            delete _transactions[i];
+
+            require(
+                exec(transaction.to, transaction.value, transaction.data, transaction.operation),
+                "Failed to execute the transaction"
+            );
+            emit TransactionExecuted(_proposalId, i);
+        }
+    }
+
+    function deleteProposal(uint256 _proposalId) public onlyOwner {
+        // Delete a proposal that governance decided not to execute.
+        delete proposalHashes[_proposalId];
+        emit ProposalDeleted(_proposalId);
+    }
+
+    function deleteRejectedProposal(
+        uint256 _proposalId,
+        uint32 _originalTime,
+        bytes memory _ancillaryData,
+        SkinnyOptimisticOracleInterface.Request memory _request
+    ) public {
+        // This will revert if the price has not settled.
+        (, int256 price) =
+            skinnyOptimisticOracle.settle(address(this), identifier, _originalTime, _ancillaryData, _request);
+
+        // Check that proposal was rejected.
+        require(price == 0, "Proposal was not rejected");
```

Comment:
Should this instead check that it's `!= 1`? I would think that anything that's _not accepted_ is rejected, right?